dandi/dandiapi.py

```diff
@@ -518,3 +518,5 @@ def get_asset_bypath(self, dandiset_id, version, asset_path):
             )
         except ValueError:
             return None
+        else:
+            return asset
```

Comment:
Just for my own sake: what is the benefit from having this in else instead of at the end of the try/except block or even just after it?

Reply:
The line should be outside of the `try:` block because the `try:` is only supposed to catch failures in the `(asset,)` assignment (True, a `return` can't raise an exception, so it makes no practical difference here, but placing the line outside of the `try:` is in keeping with the practice of making `try:` enclose only what you want to catch errors from and nothing else.)  As for why to put it under `else` instead of just unindenting it a level, that's largely a stylistic choice, a consequence of my preference for writing things like:

```
if x:
    return y
else:
    return z
```

instead of:

```
if x:
    return y
return z
```

Reply:
gotcha, thanks.  Just didn't want to miss some unknown to me property.  I should use `try/else` more often I think